src/identities/providers/github/GitHubProvider.ts

```diff
@@ -280,102 +280,92 @@ class GitHubProvider extends Provider {
+          if (response.status === 401) {
+            throw new identitiesErrors.ErrorProviderUnauthenticated(
+              `Invalid access token`,
+            );
+          }
+          throw new identitiesErrors.ErrorProviderCall(
+            `Provider responded with ${response.status} ${response.statusText}`,
           );
         }
-        throw new identitiesErrors.ErrorProviderCall(
-          `Provider responded with ${response.status} ${response.statusText}`,
-        );
-      }
-      let data;
-      try {
-        data = await response.json();
-      } catch (e) {
-        throw new identitiesErrors.ErrorProviderCall(
-          `Provider response body is not valid JSON`,
-          { cause: e },
-        );
-      }
-      for (const item of data) {
-        const identityData = await this.getIdentityData(
-          authIdentityId,
-          item.login,
-        );
-        if (
-          identityData &&
-          identitiesUtils.matchIdentityData(identityData, searchTerms)
-        ) {
-          yield identityData;
+        let data;
+        try {
+          data = await response.json();
+        } catch (e) {
+          throw new identitiesErrors.ErrorProviderCall(
+            `Provider response body is not valid JSON`,
+            { cause: e },
+          );
         }
-      }
-      if (data.length === 0) {
-        break;
-      } else {
-        pageNum = pageNum + 1;
-      }
-    }
-    pageNum = 1;
-    while (true) {
-      const request = this.createRequest(
-        `${this.apiUrl}/user/followers?per_page=100&page=${pageNum}`,
-        {
-          method: 'GET',
-        },
-        providerToken,
-      );
-      const response = await fetch(request);
-      if (!response.ok) {
-        if (response.status === 401) {
-          throw new identitiesErrors.ErrorProviderUnauthenticated(
-            `Invalid access token`,
+        const error = data?.errors?.at?.(0);
+        if (error != null) {
+          throw new identitiesErrors.ErrorProviderCall(
+            `Provider response body contains an error: ${error.message}`,
+            {
+              data: error
+            }
           );
         }
-        throw new identitiesErrors.ErrorProviderCall(
-          `Provider responded with ${response.status} ${response.statusText}`,
-        );
-      }
-      let data;
-      try {
-        data = await response.json();
-      } catch (e) {
-        throw new identitiesErrors.ErrorProviderCall(
-          `Provider response body is not valid JSON`,
-          { cause: e },
-        );
-      }
-      for (const item of data) {
-        const identityData = await this.getIdentityData(
-          authIdentityId,
-          item.login,
-        );
-        if (
-          identityData &&
-          identitiesUtils.matchIdentityData(identityData, searchTerms)
-        ) {
-          yield identityData;
+        const foundIdentityData: any[] = data?.data?.user?.[identityGroup]?.nodes ?? [];
```

Comment:
How robust is this? And why `any` type?

Reply:
I decided to not use the IdentityData type here, because it doesn't contain all the required properties, which I add in when I iterate over the array. The ternery operators are used in case the server returns an error data structure, but realistically, as errors are caught before this line, `nodes` should always be an array and not undefined.

Reply:
fixed
```
 // FollowerConnection and FollowingConnection always exists on User
        const foundIdentityGroupData = data.data.user[identityGroup];
        // Array<User> always exists on FollowerConnection and FollowingConnection
        const foundIdentityData: IdentityData[] = foundIdentityGroupData.nodes;
```